### PR TITLE
Automate updating `CHANGELOG.md` file

### LINKED ISSUE #563: Automation for Changelog
### Summary of the new feature / enhancement

We should create automation to generate our changelog based on PRs

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by alyssa1303**
# PR Summary

This PR create a `tool` folder which contains a `setupReleaseTools.ps1` file to install and set up [PowerShellforGitHub](https://github.com/microsoft/PowerShellForGitHub) module and a `releaseTools.psm1` file to update the CHANGELOG.md file and create a draft PR for the file

## PR Context

Resolve #563 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]
LGTM

**Review by alerickson — APPROVED** [review]

File: tool/releaseTools.psm1 [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        $CurrentChangeLog[0..1]
+        Set-Changelog $Version
+        $CurrentChangeLog[2..$CurrentChangeLog.Length]
+    ) | Set-Content -Encoding utf8NoBOM -Path $script:ChangelogFile
+}
+
+<#
+.SYNOPSIS
+    Updates the PowerShellGet.psd1 file
+# TODO: Update ModuleVersion and Prerelease once the format is agreed upon
```
Comment (anamnavi) [inline tool/releaseTools.psm1]:
```suggestion
# Note: Update ModuleVersion and Prerelease after this, manually.
```

File: tool/releaseTools.psm1 [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+function Remove-Release-Label {
+    $script:PullRequests = $script:Repo | Get-GitHubPullRequest -State 'closed' |
+        Where-Object { $_.labels.LabelName -match 'Release' } |
+        Where-Object { -not $_.title.StartsWith("[WIP]") } | 
+        Where-Object { -not $_.title.StartsWith("WIP") }
+
+    $script:PullRequests | ForEach-Object {
+        $script:Repo | Remove-GitHubIssueLabel -Label Release -Issue $_.PullRequestNumber
+    }
+}
```
Comment (anamnavi) [inline tool/releaseTools.psm1]: nitpick to add newline:
```suggestion
}

```

File: tool/setupReleaseTools.ps1 [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Write-Host "Install and import PowerShell modules"
+Set-PSResourceRepository -Name PSGallery -Trusted | Out-Null
+Install-Module -Name PowerShellForGitHub -Scope CurrentUser -Force
+Import-Module $PSScriptRoot/releaseTools.psm1
+
+Write-Host "Setup authentication"
+Set-GitHubConfiguration -SuppressTelemetryReminder
+$password = ConvertTo-SecureString -String $Token -AsPlainText -Force
+Set-GitHubAuthentication -Credential (New-Object System.Management.Automation.PSCredential ("token", $password))
```
Comment (anamnavi) [inline tool/setupReleaseTools.ps1]: nitpick to add newline:
```suggestion
Set-GitHubAuthentication -Credential (New-Object System.Management.Automation.PSCredential ("token", $password))

```

**Review by anamnavi — APPROVED** [review]
LGTM, thanks for setting this up! Just left some nitpick suggestions, nothing blocking.

**Comment by alerickson** [comment]
resolves #563

**Pull request merged**